dashboard/app/controllers/report_abuse_controller.rb

```diff
@@ -31,57 +31,77 @@ def protected_project?
     project_owner.permission?(UserPermission::PROJECT_VALIDATOR)
   end
 
-  def report_abuse
+  def send_abuse_report(input_name, input_email, input_age, input_abuse_url)
+    unless Rails.env.development? || Rails.env.test?
+      subject = FeaturedProject.featured_channel_id?(params[:channel_id]) ?
+        'Featured Project: Abuse Reported' :
+        'Abuse Reported'
+      response = HTTParty.post(
+        'https://codeorg.zendesk.com/api/v2/tickets.json',
+        headers: {"Content-Type" => "application/json", "Accept" => "application/json"},
+        body: {
+          ticket: {
+            requester: {
+              name: (input_name == '' ? input_email : input_name),
+              email: input_email
+            },
+            subject: subject,
+            comment: {
+              body: [
+                "URL: #{input_abuse_url}",
+                "abuse type: #{params[:abuse_type]}",
+                "user detail:",
+                params[:abuse_detail]
+              ].join("\n")
+            },
+            custom_fields: [{id: AGE_CUSTOM_FIELD_ID, value: input_age}],
+            tags: (params[:abuse_type] == 'infringement' ? ['report_abuse', 'infringement'] : ['report_abuse'])
+          }
+        }.to_json,
+        basic_auth: {username: 'dev@code.org/token', password: Dashboard::Application.config.zendesk_dev_token}
+      )
+      raise ZendeskError.new(response.code, response.body) unless response.success?
+    end
+
+    if params[:channel_id].present?
+      channel_id = params[:channel_id]
+
+      abuse_score = update_channel_abuse_score(channel_id)
+
+      update_file_abuse_score('assets', channel_id, abuse_score)
+      update_file_abuse_score('files', channel_id, abuse_score)
+    end
+  end
+
+  def report_abuse_pop_up
     unless protected_project?
+      unless verify_recaptcha || !require_captcha?
+        flash[:alert] = I18n.t('project.abuse.report_abuse_form.validation.captcha')
+        return head :forbidden
+      end
+
+      name = current_user&.name
+      email = current_user&.email
+      age = current_user&.age
+      abuse_url = CDO.studio_url(params[:abuse_url]) # reformats url
 
+      send_abuse_report(name, email, age, abuse_url)
```

Comment:
What happens if you're signed out and try to report abuse?

Reply:
Put an alternative text in for that case!